api/jobset/v1alpha2/jobset_types.go

```diff
@@ -79,6 +79,8 @@ const (
 )
 
 // JobSetSpec defines the desired state of JobSet
+// +kubebuilder:validation:XValidation:rule="!(has(self.startupPolicy) && self.startupPolicy.startupPolicyOrder == 'InOrder' && self.replicatedJobs.exists(x, has(x.dependsOn)))",message="StartupPolicy and DependsOn APIs are mutually exclusive"
```

Comment:
Can we get some kind of test that confirms this works?

Reply:
We test it here: https://github.com/kubernetes-sigs/jobset/blob/c0bf2320b45a0ece4a480e9438760763e813f9b2/test/integration/webhook/jobset_webhook_test.go#L420